examples/Android/SnowboyAlexaDemo/src/ai/kitt/snowboy/audio/RecordingThread.java

```diff
@@ -147,6 +162,24 @@ private void record() {
         if (null != listener) {
             listener.stop();
         }
+
+        // restarts recording if reset disk space was required
+        if (shouldReleaseDiskSpace()) {
+            diskSpaceCounterInBytes = 0;
+            this.record();
```

Comment:
suppose you run this program forever, then the `this.record()` function will be called repeatedly until it rans out of stack to store function calls. Then you'll have stack overflow.